packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -164,9 +177,40 @@ class OverlayEntry extends ChangeNotifier {
   ///
   /// You need to call this function if the output of [builder] has changed.
   void markNeedsBuild() {
+    assert(!_disposedByOwner);
     _key.currentState?._markNeedsBuild();
   }
 
+  void _didUnmount() {
+    if (_disposedByOwner && !mounted) {
+      super.dispose();
```

Comment:
No functionality of the superclass appears to be used in this implementation, so there is really no reason to call its dispose method... Technically, you should dispose the _overlayStateMounted, though.

Reply:
(see also the other comment about removing the super class altogether)